helix-view/src/clipboard.rs

```diff
@@ -192,6 +272,9 @@ mod provider {
     pub struct CommandProvider {
         pub get_cmd: CommandConfig,
         pub set_cmd: CommandConfig,
+        pub get_primary_cmd: Option<CommandConfig>,
+        pub set_primary_cmd: Option<CommandConfig>,
+        pub selection_buf: String,
```

Comment:
Is there any usecase for having a fallback primary selection buffer ? I think it's better to show an error that the primary clipboard is not available since it will give a false impression of it's existence.

Reply:
Actually, this way Windows and Mac users, as well as ssh users will be able to move text inside the editor. I think it's rather graceful fallback in case system clipboard is not available or cannot handle primary selection.

Reply:
We actually have [registers](https://github.com/helix-editor/helix/pull/557/files#diff-492d9e0201bf57b23dac04280ceff4f15735ac3781da61f9d0c43bc40f5d980cR5) for this exact purpose, though it's not in the user facing docs yet.

Reply:
Maybe we can make that fallback optional rather than remove it?

Reply:
I'm in favor of removing it altogether, since the use-case is solved by registers and it will probably add to confusion when used in windows and mac.